src/ja/server/database/database.py

```diff
@@ -20,6 +20,15 @@ def find_job_by_id(self, job_id: str) -> DatabaseJobEntry:
         @return The stored data about the job, or None if no such job was found.
         """
 
+    @abstractmethod
+    def find_job_by_label(self, label: str) -> List[Job]:
```

Comment:
One last thing, I believe we should also have a unit test in the Database tests for this function as well.